forte/predictor.py

```diff
@@ -0,0 +1,185 @@
+        '''Feature_resource is prodcued from train pipeline.
+        An example looks like
+        {
+        'scope': ft.onto.base_ontology.Sentence,
+        'schemes': {
+            'text_tag': {
+                'extractor': <forte.data.extractor.extractor.TextExtractor>,
+                'converter': <forte.data.extractor.converter.Converter>,
+                'type': DATA_INPUT
+            },
+
+            'char_tag': {
+                'extractor': <forte.data.extractor.extractor.CharExtractor>,
+                'converter': <forte.data.extractor.converter.Converter>,
+                'type': DATA_INPUT
+            },
+
+            'ner_tag': {
+                'extractor': <forte.data.extractor.extractor.AnnotationSeqExtractor>,
+                'converter': <forte.data.extractor.converter.Converter>,
+                'type': DATA_OUTPUT,
+                'unpadder': <forte.data.extractor.converter.SameLengthUnpadder>}
+            }
+        }
+        '''
+        self.batch_size = batch_size
+        self.feature_resource = feature_resource
+        self.cross_pack = cross_pack
+        self.pack_pools = []
+        self.instance_pools = []
+        self.features_pools = []
+
+    def current_pack_len(self) -> int:
+        return len(set(self.pack_pools))
+
+    def convert(self, feature_collections: List) -> Dict:
+        example_collection = {}
+        for features_collection in feature_collections:
+            for tag, feature in features_collection.items():
+                if tag not in example_collection:
+                    example_collection[tag] = []
+                example_collection[tag].append(feature)
+
+        tensor_collection = {}
+        for tag, features in example_collection.items():
+            converter = self.feature_resource['schemes'][tag]["converter"]
+            need_pad = self.feature_resource['schemes'][tag]["need_pad"]
+            if tag not in tensor_collection:
+                tensor_collection[tag] = {}
+            if need_pad:
+                tensor, mask = converter.convert(features)
+                tensor_collection[tag]["tensor"] = tensor
+                tensor_collection[tag]["mask"] = mask
+            tensor_collection[tag]["features"] = features
+        return tensor_collection
+
+    def yield_batch(self, pack: DataPack) -> Dict:
+        for instance in pack.get(self.feature_resource['scope']):
+            feature_collection = {}
+            for tag, scheme in self.feature_resource['schemes'].items():
+                if scheme['type'] == DATA_INPUT:
+                    extractor = scheme["extractor"]
+                    feature = extractor.extract(pack, instance)
+                    feature_collection[tag] = feature
+
+            self.pack_pools.append(pack)
+            self.instance_pools.append(instance)
+            self.features_pools.append(feature_collection)
+
+            if len(self.features_pools) == self.batch_size:
```

Comment:
So this is a fixed size batcher. 

For batcher I suggest follow our original idea here: https://github.com/asyml/forte/blob/master/forte/data/batchers.py#L69

Where the base batcher do not determine whether to yield. A specific batcher implementation determines that.